homeassistant/components/sensor/environment_canada.py

```diff
@@ -0,0 +1,158 @@
+                   'unit': 'km/h'},
+    'wind_gust': {'name': 'Wind Gust',
+                  'unit': 'km/h'},
+    'wind_dir': {'name': 'Wind Direction'},
+    'high_temp': {'name': 'High Temperature',
+                  'unit': TEMP_CELSIUS},
+    'low_temp': {'name': 'Low Temperature',
+                 'unit': TEMP_CELSIUS},
+    'pop': {'name': 'Chance of Precip.',
+            'unit': '%'},
+    'warning': {'name': 'Warning'}
+}
+
+
+def validate_station(station):
+    """Check that the station ID is well-formed."""
+    if station is None:
+        return
+    if not re.fullmatch(r'[A-Z]{2}/s0000\d{3}', station):
+        raise vol.error.Invalid('Station ID must be of the form "XX/s0000###"')
+    return station
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES.keys())):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_STATION): validate_station,
+    vol.Inclusive(CONF_LATITUDE, 'latlon'): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'latlon'): cv.longitude,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Environment Canada sensor."""
+    if config.get(CONF_STATION):
+        ec_data = ECData(station_id=config[CONF_STATION])
+    elif config.get(CONF_LATITUDE) and config.get(CONF_LONGITUDE):
+        ec_data = ECData(coordinates=(config[CONF_LATITUDE],
+                                      config[CONF_LONGITUDE]))
+    else:
+        ec_data = ECData(coordinates=(hass.config.latitude,
+                                      hass.config.longitude))
+
+    add_devices([ECSensor(sensor_type, ec_data, config.get(CONF_NAME))
+                 for sensor_type in config[CONF_MONITORED_CONDITIONS]])
+
+
+class ECSensor(Entity):
+    """Implementation of an Environment Canada sensor."""
+
+    def __init__(self, sensor_type, ec_data, platform_name):
+        """Initialize the sensor."""
+        self.sensor_type = sensor_type
+        self.ec_data = ec_data
+        self.platform_name = platform_name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self.platform_name is None:
+            return 'EC {}'.format(SENSOR_TYPES[self.sensor_type]['name'])
+
+        return 'EC {} {}'.format(
+            self.platform_name, SENSOR_TYPES[self.sensor_type]['name'])
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self.ec_data.conditions.get(self.sensor_type)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        if self.state:
+            hidden = False
+        else:
+            hidden = True
+
+        timestamp = self.ec_data.conditions.get('timestamp')
+        if timestamp:
+            updated_utc = datetime.datetime.strptime(timestamp, '%Y%m%d%H%M%S')
```

Comment:
Should be an ISO 8601 formatted timestamp.

Reply:
I added `isoformat()` to the localized object, which I think is correct.